asset/src/_kafka_clients/producer-client.ts

```diff
@@ -87,9 +101,13 @@ export default class ProducerClient extends BaseClient<kafka.Producer> {
                 // flush the messages at the end of each slice
                 if (i === endOfBatchIndex) {
                     await this._try(() => this._flush(), 'produce', 0);
+                    this._logger.debug(`End of message batch Flushing queue before writing more messages...`);
+                    currentBatchSizeInBytes = 0; // Reset the batch size counter
                 // OR at the first message
-                } else if (i % this._bufferSize === endofBufferIndex) {
+                } else if (i % this._maxBufferMsgLength === endofBufferIndex) {
+                    this._logger.debug(`Beginning of message batch Flushing queue before writing more messages...`);
```

Comment:
This message and the size message should convey the same idea, they are flushing because a condition on the queue size/length has been met.  Think about both of these messages together and make sure the messages are written to communicate to operators, these are written too much towards the implementation.